src/ripple/rpc/handlers/ValidatorInfo.cpp

```diff
@@ -0,0 +1,63 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2019 Dev Null Productions
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/misc/ValidatorKeys.h>
+#include <ripple/app/main/Application.h>
+#include <ripple/basics/base64.h>
+#include <ripple/json/json_value.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/jss.h>
+#include <ripple/rpc/Context.h>
+
+namespace ripple {
+Json::Value doValidatorInfo (RPC::Context& context)
+{
+    // return error if not configured as validator
+    if (context.app.getValidationPublicKey().empty())
+      return RPC::not_validator_error ();
+
+    Json::Value ret;
+
+    auto pk = context.app.getValidationPublicKey();
```

Comment:
nit: Would like to see more `const` used if possible. i.e. `const auto pk = ...`

Makes it clear that the variable won't be modified later (which is the case for a lot of these).